### PR TITLE
Update to Go 1.20

### PR DESCRIPTION
**Pull request by bufdev**
Need to wait for the docker images to be published.
Also depends on https://github.com/bufbuild/buf/issues/1784

### CODE REVIEW

**Comment by jchadwick-buf** [comment]
I'm suspicious of this:
```diff
-          go-version: 1.19.5
+          go-version: 1.20
```
I think it's loading Go 1.2 instead of 1.20, because YAML is fun. I think we need to add quotes to the 1.20.

**Comment by jchadwick-buf** [comment]
Locally this also fails due to https://github.com/golangci/golangci-lint/issues/3420, so I assume that will hit us as well.

**Comment by mfridman** [comment]
We can try and bump golanci to the latest version [`v1.51.0`](https://github.com/golangci/golangci-lint/releases/tag/v1.51.0) to resolve that issue.

https://github.com/bufbuild/buf/blob/d59feb33a081913be2817f7397e1f85287ce270c/make/go/dep_golangci_lint.mk#L12

**Comment by jchadwick-buf** [comment]
Locally, that works for me and after doing so, `make all` passes. I think #1784 is invalid and not blocking this PR.

**Comment by bufdev** [comment]
Docker images are now there

File: make/buf/all.mk [hunk 1]
```diff
@@ -116,7 +116,7 @@ bufgeneratesteps:: \
 
 .PHONY: bufrelease
 bufrelease: $(MINISIGN)
-	DOCKER_IMAGE=golang:1.19.5-bullseye bash make/buf/scripts/release.bash
+	DOCKER_IMAGE=golang:1.20-bullseye bash make/buf/scripts/release.bash
```
Comment (bufdev) [inline make/buf/all.mk:119]: There is a `1.20.0-bullseye` as well, and we traditionally have locked to micros, but with our workflows for Docker, I don't think this actually makes as much sense. We may want to just lock to minors for Docker going forward, so we always push the latest.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: setup-go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19.5
+          go-version: '1.20'
```
Comment (bufdev) [inline .github/workflows/ci.yaml:17]: We may want to update this to `1.20.x`

**Review by jchadwick-buf — APPROVED** [review]
LGTM as far as I can tell. Seems to work fine locally.

**Comment by bufdev** [comment]
We're going to need to run this on a Windows box to figure out what is going on, I've re-run the job a few times and it appears as though there is an actual issue. On glancing at the 1.20 release docs (including changes to the stdlib), it's not immediately apparent of what change in 1.20 would have caused this.

**Pull request merged**